src/edge/tests/test_recombine_annotations.py

```diff
@@ -285,3 +286,189 @@ def test_adds_annotations_correctly_with_reverse_cassette(self):
         self.assertEquals(a.feature.strand, 1)
         self.assertEquals(a.base_first, len(self.upstream) + len(self.front_bs) + 1)
         self.assertEquals(a.base_last, len(self.upstream) + len(self.front_bs) + len(self.middle))
+
+    def test_preserves_annotations_on_homology_arm_fwd_that_have_not_changed(self):
+        cassette = ''.join([self.front_bs, self.back_bs])
+
+        # add annotaiton on upstream arm
+        self.fragment.annotate(len(self.upstream) + 2,
+                               len(self.upstream) + len(self.front_bs),
+                               "Up arm", "feature", 1)
+
+        c = recombine(self.genome, cassette, self.arm_len)
+        f = c.fragments.all()[0].indexed_fragment()
+
+        annotations = f.annotations()
+        self.assertEquals(len(annotations), 2)
+
+        a = annotations[0]
+        self.assertEquals(a.feature.type, 'operation')
+
+        a = annotations[1]
+        self.assertEquals(a.feature.name, 'Up arm')
+        self.assertEquals(a.feature.type, 'feature')
+        self.assertEquals(a.feature.strand, 1)
+        self.assertEquals(a.base_first, len(self.upstream) + 2)
+        self.assertEquals(a.base_last, len(self.upstream) + len(self.front_bs))
+
+    def test_preserves_annotations_on_homology_arm_rev_that_have_not_changed(self):
+        cassette = ''.join([self.front_bs, self.back_bs])
+
+        # add annotaiton on reverse strand of downstream arm
+        self.fragment.annotate(
+            len(self.upstream) + len(self.front_bs) + len(self.middle) + 1,
+            len(self.upstream) + len(self.front_bs) + len(self.middle) + len(self.back_bs),
+            "Down arm", "feature", -1)
+
+        c = recombine(self.genome, cassette, self.arm_len)
+        f = c.fragments.all()[0].indexed_fragment()
+
+        annotations = f.annotations()
+        self.assertEquals(len(annotations), 2)
+
+        a = annotations[0]
+        self.assertEquals(a.feature.type, 'operation')
+
+        a = annotations[1]
+        self.assertEquals(a.feature.name, 'Down arm')
+        self.assertEquals(a.feature.type, 'feature')
+        self.assertEquals(a.feature.strand, -1)
```

Comment:
oh. I think I know now...1 means forward strand and -1 means reverse strand.

Reply:
why do we use a number instead of a descriptive string to represent the forward and reverse strand?

Reply:
It's the BioPython and GFF convention.

Reply:
Got it! Thanks for explaining.